### DISCUSSION TITLE
css modifier only partialy supported?

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#7641] [Ideas]
It seems really weird that tailwind support partial css states, for example there is `first-line` and `first-letter` modifier, but no modifier for last-letter or last-line. In a similar way there is no option for first word, this seems really mess-up. why would you support first line but not last line and so on

can we improve support for these modifiers, and is there a workaround to make last word modifier work in tailwind css
```
:first-child        :first-of-type        :only-child
:last-child         :last-of-type         :only-of-type                  
:nth-child          :nth-of-type    
:nth-last-child     :nth-last-of-type

::first-letter      ::first-line          ::first-word
::last-letter       ::last-line           ::last-word
::nth-letter        ::nth-line            ::nth-word
::nth-last-letter   ::nth-last-line       ::nth-last-word 
```

Thanks

**Comment by adamwathan** [comment]
Hey! As far as I can tell there is no such thing as `last-letter`, `last-line`, `first-word`, `nth-word`, `last-word`, `nth-line`, `nth-last-line`, `nth-last-word`, `nth-letter`, or `nth-last-letter` in CSS, here is a complete reference of pseudo-elements:

https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements#index

Most of the others we already support:

https://tailwindcss.com/docs/hover-focus-and-other-states#quick-reference

The only ones we don't are `nth-child` with arbitrary numbers (we do support `odd` and `even`), `nth-of-type`, and `nth-last-of-type`, mainly because there's no mechanism in Tailwind yet for providing `n` as part of the modifier name.

It's trivial to add any custom variants you need though:

https://tailwindcss.com/docs/plugins#adding-variants

We will likely add more `nth-*` support when we add support for modifier arguments, like `nth[165]:bg-black` for `:nth-child(165)`.
